homework7/hw2.py

```diff
@@ -0,0 +1,38 @@
+"""
+Given two strings. Return if they are equal when both are typed into
+empty text editors. # means a backspace character.
+
+Note that after backspacing an empty text, the text will continue empty.
+
+Examples:
+    Input: s = "ab#c", t = "ad#c"
+    Output: True
+    # Both s and t become "ac".
+
+    Input: s = "a##c", t = "#a#c"
+    Output: True
+    Explanation: Both s and t become "c".
+
+    Input: a = "a#c", t = "b"
+    Output: False
+    Explanation: s becomes "c" while t becomes "b".
+
+"""
+
+
+def backspace_compare(first: str, second: str):
+    s1 = [
+        "" if char == "#" or first.index(char) == first.find("#", first.index(char)) - 1
+        # I left this condition because first[first.index(char) + 1] == '#' gives an IndexError
```

Comment:
than you can use `for i, char in enumerate(first)`, so you would not call .index() function each iteration